CMakeLists.txt

```diff
@@ -106,6 +106,18 @@ if(ENABLE_CCACHE AND (CMAKE_GENERATOR STREQUAL "Xcode"))
 	set(CMAKE_XCODE_ATTRIBUTE_LDPLUSPLUS 	"${CMAKE_BINARY_DIR}/xcode/launch-cxx")
 endif()
 
+option(ENABLE_EXPORT_COMPILE_COMMANDS "Enable export of 'compile_commands.json', used by C++ tooling such as clangd or cppcheck" ON)
```

Comment:
most devs don't need it, set it to OFF

Reply:
You sure? I mean any developer that would open up VCMI in any text editor that uses clangd would have to explicitly enable exporting compile_commands.json. Currently with this PR clangd works out of the box without any manual intervention. 

I've never encountered any problems with this export and I think the cost&risk of enabling it by default is low.